### ISSUE TITLE
vim.statusBarColorControl changes cannot be undone

### ISSUE THREAD

**Issue by ls42** [VSCodeVim/Vim#6907]
**Describe the bug**
Activated this setting test-wise, but after deactivating it the color changes stick and I cannot figure out how to undo them (tried uninstalling the vim plugin, the theme that I'm using and of course restarting the editor). 

**To Reproduce**

1. Set setting to true
2. Go into insert mode
3. Leave insert mode
4. Set value to false
5. Status bar color sticks now, even when changing to other color themes

**Expected behavior**
After undoing the setting it should go back to the regular color.

**Environment (please complete the following information):**

- Extension (VsCodeVim) version:
- VSCode version: v1.21.5
- OS: macOS Catalina 11.5

**Comment by J-Fields** [comment]
As a workaround, you can remove `statusBar.background`, `statusBar.noFolderBackground`, and `statusBar.debuggingBackground` from `workbench.colorCustomizations`

**Comment by J-Fields** [comment]
Yeah frankly I hate that this setting exists lol - it was created before I got involved with the project and because it modifies your settings (unfortunately the only way possible AFAIK), it's slow and janky. Hard to remove without making people mad though 🙂

With that said, I should spend the time to fix this issue and probably optimize it a bit...

**Comment by ls42** [comment]
I mean, it is kind of a cool feature, that's why I tried it out; I was scanning through all possible settings and found it interesting. Maybe a note to the user in the default settings would be useful.

But changing the settings every time one leaves and enters insert mode seems so... wrong :-) What does that mean for the settings sync that's included in vscode?

**Comment by J-Fields** [comment]
> What does that mean for the settings sync that's included in vscode?

This is not something I've thought about. I *really* hope it doesn't sync every time... something to look into

**Comment by GitPopcorn** [comment]
At lease, I mean, you can add the way to undone in the description of configuration `vim.statusBarColorControl`, I see you add a performance note on it, and I think another note is just fine for it.
